packages/gatsby-theme/src/components/Share/ShareFacebook.js

```diff
@@ -0,0 +1,24 @@
+import PropTypes from "prop-types"
+import { FacebookShareButton } from "react-share"
+import Icon from "../Icon"
+
+const ShareFacebook = ({ title, to, children, ...props }) => {
+  return (
+    <FacebookShareButton url={to} title={title} {...props} resetButtonStyle={false}>
+      <Icon icon='facebook' />
+      {children}
+    </FacebookShareButton>
+  )
+}
```

Comment:
We should maybe do some refactoring in the internals of this component and add a single file with a Hashmap that receives an type and returns the appropriate component from the react-share library.